pkg/portal/security_test.go

```diff
@@ -351,7 +351,12 @@ func TestSecurity(t *testing.T) {
 				// [2] https://go.googlesource.com/go/+/go1.16.2/src/net/http/fs.go#337
 				if tt.name == "/" || tt.name == "/main.js" {
 					err = testpoller.Poll(1*time.Second, 5*time.Millisecond, func() (bool, error) {
-						return len(auditHook.AllEntries()) == 1, nil
+						if len(auditHook.AllEntries()) == 1 {
+							if _, ok := auditHook.AllEntries()[0].Data[audit.MetadataPayload]; ok {
+								return true, nil
+							}
```

Comment:
Just for interest, in what scenarios `len(AllEntries)` is 1 and MetatadaPayload is not there? what data is there?

Reply:
No payload:
```
ok: false
entry.data is map[adminOp:true createdTime:2021-07-01T15:58:27Z logKind:ifxaudit source:aro-admin]
```

Payload:
```
ok: true
entry.data is map[adminOp:true createdTime:2021-07-01T15:58:27Z logKind:ifxaudit payload:{"env_ver":2.1,"env_name":"#Ifx.AuditSchema","env_time":"2021-07-01T15:58:27Z","env_epoch":"25b6a65a-adc8-47ff-88b8-a4e91de1604c","env_seqNum":4,"env_flags":257,"env_appId":"aro-admin","env_cloud_name":"AzurePublicCloud","env_cloud_role":"rp","env_cloud_roleInstance":"testhost","env_cloud_environment":"AzurePublicCloud","env_cloud_location":"eastus","env_cloud_ver":1,"CallerIdentities":[{"CallerIdentityType":"Username","CallerIdentityValue":"","CallerIpAddress":"bufferedpipe"}],"Category":"ResourceManagement","OperationName":"GET /main.js","Result":{"ResultType":"Fail","ResultDescription":"Status code: 403"},"requestId":"","TargetResources":[{"TargetResourceType":"","TargetResourceName":"/main.js"}]} source:aro-admin]
```

Reply:
But what this happens?

It feels like we are fixing a sympthom (test), not the root cause of the issue

Reply:
There's a race condition that happens that described in the comment section there. The previous attempt to fix it, did not fully solve it. This is just to wait for the payload to show up. I agree that it's not ideal, but I'm not sure how to confirm that this event is fully completed other than just retry and see if payload is present. Do you have any suggestions?

Reply:
Right, I did not see the comment outside of the PR diff, my bad. I think it is a good enough explanation and I'm ok with it.